include/lib/map.h

```diff
@@ -4,24 +4,39 @@
 
 #include "stdlib/stdtypes.h"
 
-/** Data structure for a generic map */
+/** Structures for a generic map. */
+struct entry {
+    /** The key associated with the entry. */
+    void *key;
+    /** The value associated with the entry. */
+    void *value;
+};
+
 struct map {
-    //TODO
+    /** Table which holds all entries. */
+    struct entry *table;
+    /** Amount of space used. */
+    size_t used;
+    /** Maximum space to allocate for entries */
+    size_t size;
 };
 
-/** Initializes a new hashtable */
-void map_init(struct map *m);
+/** Initializes a new hashmap. Returns `-1` if not allocated. */
+int map_init(struct map *m, size_t size);
```

Comment:
The return type of this should be void, not int, in order to align with the other standard library functions.